contracts/CompoundProtocolPriceView.sol

```diff
@@ -0,0 +1,91 @@
+pragma solidity ^0.6.6;
+pragma experimental ABIEncoderV2;
+
+import "./AnchoredPriceView.sol";
+
+interface CompoundProtocolPriceViewI {
+    function isPriceOracle() external pure returns (bool);
+
+    function getUnderlyingPrice(address cToken) external view returns (uint256);
+}
+
+contract CompoundProtocolPriceView is AnchoredPriceView, CompoundProtocolPriceViewI {
+
+    /**
+     * @param data_ Address of the Oracle Data contract
+     * @param source_ The reporter address whose price will be used if it matches the anchor
+     * @param anchor_ The PriceOracleProxy that will be used to verify source price, or serve prices not given by the source
+     * @param anchorToleranceMantissa_ The tolerance allowed between the anchor and median. A tolerance of 10e16 means a new median that is 10% off from the anchor will still be saved
+     * @param tokens_ The CTokens struct that contains addresses for CToken contracts
+     */
+    constructor(OpenOraclePriceData data_,
+                address source_,
+                address anchor_,
+                uint anchorToleranceMantissa_,
+                CTokens memory tokens_) public AnchoredPriceView(data_, source_, anchor_, anchorToleranceMantissa_, tokens_) {
+    }
+
+    /**
+     * @notice Flags that this contract is meant to be compatible with Compound v2 PriceOracle interface.
+     * @return true, this contract is meant to be used as Compound v2 PriceOracle interface.
+     */
+    function isPriceOracle() external pure override returns (bool) {
+        return true;
+    }
+
+    /**
+     * @notice Implements the method of the PriceOracle interface of Compound v2.
+     * @dev converts from 1e6 decimals of Open Oracle to 1e(36 - underlyingDecimals) of PriceOracleProxy
+     * @param cToken The cToken address for price retrieval
+     * @return The price for the given cToken address
+     */
+    function getUnderlyingPrice(address cToken) external view override returns (uint256) {
+        uint256 priceSixDecimals;
+
+        if(cToken == cUsdcAddress || cToken == cUsdtAddress) {
+            priceSixDecimals = oneDollar;
+        } else if (cToken == cWbtcAddress || cToken == cEthAddress) {
+            priceSixDecimals = _prices[getOracleKey(cToken)];
+        } else {
+            uint256 usdPerEth = _prices["ETH"];
+            uint256 ethPerToken = anchor.getUnderlyingPrice(cToken);
```

Comment:
the fact that this contract still has only this one reference to the anchor is a good indicator that writing anchor prices to storage like in the other PR is the correct abstraction. Good thinking. 

Too bad though, I don't think there is a way to pull *all* the cToken references through into this ocntract b/c of needing to know the cusdc address

Reply:
Yes, having tokens addresses shared between these 2 contracts is messing up this split. I kind of like splitting anchor that happens to be protocol oracle proxy from usage by protocol proxy, but it definitely needs more thought process.